api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImpl.java

```diff
@@ -42,6 +45,27 @@
 	@Autowired
 	private ProvenanceTranslator<org.openmrs.Encounter> provenanceTranslator;
 	
+	private Coding mapLocationToClass(Location location) {
+		Coding coding = new Coding();
+		coding.setSystem(FhirConstants.ENCOUNTER_CLASS_VALUE_SET_URI);
+		// The default code for anything that cannot be matched with FHIR codes.
+		coding.setCode("AMB");
+		if (location == null) {
+			return coding;
+		}
+		// TODO: These are a subset of locations in the Ref. App.; figure out a more generic way for
+		// this mapping, possibly by including encounter type too.
+		switch (location.getName()) {
+			case "Inpatient Ward":
+				coding.setCode("IMB");
+				break;
+			case "Outpatient Clinic":
+				coding.setCode("AMB");
+				break;
+		}
+		return coding;
```

Comment:
Rather than driving this by location, the right thing to drive it by, it seems to me, is the visit the encounter is part of. I don't think the location hierarchy in the RefApp is of much use. We also might want to use a configuration file similar to the ObservationCategoryMap to make this determination.

Reply:
Thanks for the suggestion; yes using a configurable map makes sense. But did you mean to actually rely on Obs.Concept of OpenMRS to figure out the FHIR Encounter class? Looking at the [V3 Value SetActEncounterCode list](https://www.hl7.org/fhir/v3/ActEncounterCode/vs.html) it feels it is mostly about location, no? I would definitely prefer to rely on a coded concept than location strings (which I am not even if they are commonly used outside Ref. App.) but I don't know what that concept should be.

If you want, I can loop through all concepts of all observations in the encounter and come up with some heuristics (do we have any repository of sample databases for various implementations)?

Reply:
I wish we had sample databases to do that! I agree that the location is likely to helpful in determining this. "Inpatient Ward" and "Outpatient Clinic" are definitely not widely-used... If nothing else, we have a significant number of installs in non-English speaking places. I wouldn't suggest relying on someone storing an Obs. The encounter location is probably the best we can do.

Reply:
@ibacher I changed this based on what we agreed on, i.e., added a location to class properties file. Please note that I pushed two commits, one is just moving some files and removing a redundant interface and the last one is the actual code changes since last time. PTAL and thanks.